include/EntryPointConfirmOptIn.php

```diff
@@ -43,40 +43,104 @@
 /**
  * Class confirm_opt_in
  */
-class ConfirmOptIn
+class EntryPointConfirmOptIn
 {
     /**
      * @var EmailAddress $emailAddress
      */
     private $emailAddress;
+    
+    /**
+     * 
+     * @param array $request
+     * @param array $post
+     */
+    public function __construct($request = null, $post = null) {
+        
+        if(is_null($request)) {
+            $request = $_REQUEST;
+        }
+        
+        if(is_null($post)) {
+            $post = $_POST;
+        }
+        
+        $method = isset($request['method']) && $request['method'] ? $request['method'] : null;
+        
+        switch($method) {
+            case 'confirmOptInSelected':
+                $output = $this->methodConfirmOptInSelected($post);
+                break;
+            default:
+                $output = $this->methodConfirmOptInUser($request);
+                break;
+        }
+        
+        echo $output;
+        sugar_cleanup();
+    }
+    
+    /**
+     * 
+     * @global array $sugar_config
+     * @param array $post
+     * @return string|boolean
+     */
+    private function methodConfirmOptInSelected($post) {
```

Comment:
naming a method with the prefix method seems a little redundant.

Reply:
i mean that if you are calling a method - why do you need to name it method?

in the same way if you are instatating a class (eg DogClass). Why would you put "class" in the name of the class. It's not a big thing. It's just unnecessary.

If we are making this a controller, the convention in SuiteCRM is to prefix it with "action".

Reply:
you can, but the "action" already in use. It's also came from the request: https://github.com/salesagility/SuiteCRM/pull/4952/files#diff-1ca841fc98ed932c05fcbe4880f3f6e8R68